File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.*;
class Solution {
    public int[] solution(int n) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int i = 2; i <= n; i++) {
            while (n % i == 0) {
                if (!result.contains(i)) {
                    result.add(i);
                }
                n /= i;
            }
        }
        
        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120852. 소인수분해/소인수분해.js
function solution(n) {
    let answer = [];
    for(let i=2; i<=n; i++){
        if(n % i == 0){
            let cnt = 0
            for(let j=2; j<=i; j++){
                if(i % j == 0){
                    cnt++;
                }
            }
            if(cnt <=1) answer.push(i); 
        } 
    }
    console.log(answer)
    return answer;
}